### ISSUE TITLE
Srcset string returns unwanted images

### ISSUE THREAD

**Issue by tomsterritt** [ResponsiveImagesCG/wp-tevko-responsive-images#48]
For example, with the following sizes set:

``` php
set_post_thumbnail_size(120, 100);
add_image_size('thumb-m', 180, 150);
add_image_size('thumb-l', 240, 200);
add_image_size('thumb-xl', 390, 325);
```

Passing 'thumb-l' as 'theLargestImageSizeNeeded' returns a list containing all image sizes, with the thumbnail size in twice.

I'm expecting only the following sizes: thumbnail, thumb-m, thumb-l - is that not correct?

**Comment by tevko** [comment]
hey @tomsterritt would you mind replying with the generated img markup?

**Comment by tomsterritt** [comment]
My template:
``` php
echo '<img src="..." '.tevkori_get_src_sizes(get_post_thumbnail_id($post->ID), 'thumb-l').' width="120" height="100">';
```

The output (full URL/filenames removed):
``` html
<img src="/wp-content/uploads/2014/12/filename-120x100.jpg" 
srcset="/wp-content/uploads/2014/12/filename-120x100.jpg 120w, 
/wp-content/uploads/2014/12/filename-120x100.jpg 120w, 
/wp-content/uploads/2014/12/filename-180x150.jpg 180w, 
/wp-content/uploads/2014/12/filename-240x200.jpg 240w, 
/wp-content/uploads/2014/12/filename-390x325.jpg 390w, 
/wp-content/uploads/2014/12/filename.jpg 480w" 
width="120" height="100">
```

**Comment by tevko** [comment]
My guess is that the default images are happening because ``set_post_thumbnail_size`` also creates a   new image size, or at least wordpress thinks that is what's happening. http://codex.wordpress.org/Function_Reference/set_post_thumbnail_size

As for the unwanted image sizes, it looks like we're gonna have to rethink how that functionality works for theme developers. The intended functionality is that you get back all sizes under the largest size requested, but that doesn't make much sense when (if it existed) a 5x resolution screen is requesting a thumbnail image. Right now those extra sizes won't hurt you or even be requested if you have a proper sizes attribute on the img tag

**Comment by tomsterritt** [comment]
Ah yes, you're right, thanks. I hadn't regenerated the images to remove it.

**Comment by joemcgill** [comment]
We should be able to check for duplicate image URLs and remove duplicates from the array of candidates just in case. For the second issue, we could either provide a filter hook for affecting the image sizes that are included by `tevkori_get_srcset_string()` or we could add a third `$args` parameter to the function that accepts include and exclude rules.

**Comment by side777** [comment]
a hook would enable us to manipulate the sizes used in the post editor. i would prefer this. 

> Right now those extra sizes won't hurt you
they may significantly increase the size of the html...

**Comment by tomsterritt** [comment]
A hook sounds better for use in the post editor, but how would it work for templates where you want to apply different filters depending on how the image is used? For a thumbnail you'd probably want to filter all the larger images, whereas within a post you might not really need the smallest ones.

**Comment by joemcgill** [comment]
I'd like to pick this up in version 2.3. I'm thinking we would handle this by passing a third `$args` parameter to `tevkori_get_srcset_array()` and also to `tevkori_get_srcset_string()` which would allow developers to pass a set of sizes to either include or exclude from the srcset. If include is used, only the size passed to the `$size` param and those which have been included will return (if they exist).

*Example:*

```
$include = array( 'my-custom-size-1', 'my-custom-size-2' );
echo tevkori_get_srcset_string( $id, 'medium', array( 'include'  => $include ) );
```

This would print 
```
srcset="/path/to/medium {{width}}w, 
  /path/to/my-custom-size-1 {{width}}w, 
  /path/to/my-custom-size-2 {{width}}w"
```

If exclude is used, all valid sizes that match the aspect ratio of the image requested via the `$size` parameter would be returned—except for those which have been excluded.

*Example:* assuming we would normally get medium, large, custom-1, and full sized images.

```
$exclude = array( 'full' );
// or optionally $exclude = 'full';
echo tevkori_get_srcset_string( $id, 'medium', array( 'exclude'  => $exclude ) );
```

This would print 
```
srcset="/path/to/medium {{width}}w, 
  /path/to/large {{width}}w, 
  /path/to/custom-1 {{width}}w"
```

The args should also be filterable so these could be set globally in functions files, rather than on a case by case basis in templates.

**Comment by side777** [comment]
^^ great!

**Comment by jaspermdegroot** [comment]
@tomsterritt 

> I'm expecting only the following sizes: thumbnail, thumb-m, thumb-l - is that not correct?

Like @tevko said ```set_post_thumbnail_size``` also creates an image size. The name of that image size is "post-thumbnail". I thought it might be useful to mention that because the ```set_post_thumbnail_size``` documentation doesn't.
Image size "thumbnail" (and its alias "thumb") is always the image size that has been set in the Settings > Media menu.

**Comment by joemcgill** [comment]
This should be fixed as of 3346000.
